File: backend/src/controllers/contatoController.js
import { ContatoRepository } from '../repositories/contatoRepository.js';

const contatoRepo = new ContatoRepository();

export async function postContato(req, res, next) {
  try {
    const { nomeClinica, nomeContato, email, telefone, cidade, mensagem } = req.body || {};
    if (!nomeClinica?.trim() || !nomeContato?.trim() || !email?.trim() || !telefone?.trim()) {
      const err = new Error('Preencha nome da clínica, seu nome, e-mail e telefone.');
      err.status = 400;
      throw err;
    }
    const em = String(email).trim().toLowerCase();
    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(em)) {
      const err = new Error('E-mail inválido.');
      err.status = 400;
      throw err;
    }
    const out = await contatoRepo.create({
      nomeClinica: String(nomeClinica).trim(),
      nomeContato: String(nomeContato).trim(),
      email: em,
      telefone: String(telefone).replace(/\D/g, ''),
      cidade: cidade ? String(cidade).trim() : null,
      mensagem: mensagem ? String(mensagem).trim() : null,
    });
    res.status(201).json({ ok: true, id: out.id, message: 'Recebemos seu contato. Em breve nossa equipe retorna.' });
  } catch (e) {
    next(e);
  }
}

File: backend-java/src/main/java/com/clinica/bot/controller/ContatoController.java
package com.clinica.bot.controller;

import com.clinica.bot.domain.Contato;
import com.clinica.bot.exception.ApiException;
import com.clinica.bot.repository.ContatoRepository;
import com.clinica.bot.util.CpfUtils;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;
import java.util.regex.Pattern;

@RestController
@RequiredArgsConstructor
public class ContatoController {
    private static final Pattern EMAIL = Pattern.compile("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$");
    private final ContatoRepository contatos;

    @PostMapping("/contato")
    @ResponseStatus(HttpStatus.CREATED)
    public Map<String, Object> criar(@RequestBody Map<String, Object> body) {
        String clinica = text(body.get("nomeClinica")), nome = text(body.get("nomeContato")), email = text(body.get("email"));
        String telefone = CpfUtils.digitsOnly(text(body.get("telefone")));
        if (clinica.isBlank() || nome.isBlank() || email.isBlank() || telefone.isBlank()) throw new ApiException("Preencha nome da clínica, seu nome, e-mail e telefone.", 400);
        email = email.toLowerCase();
        if (!EMAIL.matcher(email).matches()) throw new ApiException("E-mail inválido.", 400);
        Map<String, Object> data = new HashMap<>(body);
        data.put("nomeClinica", clinica); data.put("nomeContato", nome); data.put("email", email); data.put("telefone", telefone);
        Contato contato = contatos.create(data);
        return Map.of("ok", true, "id", contato.getId(), "message", "Recebemos seu contato. Em breve nossa equipe retorna.");
    }
    private static String text(Object value) { return value == null ? "" : String.valueOf(value).trim(); }
}
